Directory.Build.props

```diff
@@ -13,8 +13,8 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <CodeAnalysisRuleSet Condition="'$(OS)' == 'Windows_NT'">$(SolutionDir)\BotBuilder-DotNet.ruleset</CodeAnalysisRuleSet>
-    <CodeAnalysisRuleSet Condition="'$(OS)' != 'Windows_NT'">$(SolutionDir)/BotBuilder-DotNet.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSet Condition="'$(OS)' == 'Windows_NT'">$(MSBuildThisFileDirectory)\BotBuilder-DotNet.ruleset</CodeAnalysisRuleSet>
```

Comment:
>MSBuildThisFileDirectory [](start = 63, length = 24)

But this isn't where the ruleset is.  This will not resolve the ruleset.  Have you actually tried this?

Reply:
Yes. I tried this.

I suppose `$(MSBuildThisFileDirectory)\BotBuilder-DotNet.ruleset` should be the ruleset file in the same directory as Directory.Build.props? (aka in the solution root)

Reference 0: I moved DBP to other places to verify this and notice that rebuild will not clear ruleset preference, one should remove bin & obj manually

Reference 1: https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-reserved-and-well-known-properties?view=vs-2019